Windows 11 ARM build broken

Some runner image update appears to have broken the build

monitor https://github.com/actions/partner-runner-images/issues

```
2025-08-05T20:16:35.7693347Z Current runner version: '2.327.1'
2025-08-05T20:16:35.7724499Z ##[group]Runner Image Provisioner
2025-08-05T20:16:35.7725576Z Hosted Compute Agent
2025-08-05T20:16:35.7726117Z Version: 20250711.363
2025-08-05T20:16:35.7726670Z Commit: 6785254374ce925a23743850c1cb91912ce5c14c
2025-08-05T20:16:35.7727406Z Build Date: 2025-07-11T20:04:25Z
2025-08-05T20:16:35.7728226Z ##[endgroup]
2025-08-05T20:16:35.7728811Z ##[group]VM Image
2025-08-05T20:16:35.7729499Z - OS: Windows (Arm64)
2025-08-05T20:16:35.7730087Z - Source: Partner
2025-08-05T20:16:35.7730620Z - Name: Microsoft Windows 11 Desktop by Arm Limited
2025-08-05T20:16:35.7731274Z - Version: 20250728.24.1
2025-08-05T20:16:35.7732164Z - Included Software: https://github.com/actions/partner-runner-images/blob/main/images/arm-windows-11-image.md

2025-08-05T20:16:51.9065608Z   Platform: arm64
2025-08-05T20:16:51.9066450Z   UCRTVersion: 10.0.26100.0
2025-08-05T20:16:51.9066844Z   UniversalCRTSdkDir: C:\Program Files (x86)\Windows Kits\10\
2025-08-05T20:16:51.9067389Z   VCIDEInstallDir: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\VC\
2025-08-05T20:16:51.9067928Z   VCINSTALLDIR: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\
2025-08-05T20:16:51.9068387Z   VCPKG_ROOT: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\vcpkg
2025-08-05T20:16:51.9068974Z   VCToolsInstallDir: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\
2025-08-05T20:16:51.9069623Z   VCToolsRedistDir: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Redist\MSVC\14.44.35112\
2025-08-05T20:16:51.9070082Z   VCToolsVersion: 14.44.35207
2025-08-05T20:16:51.9070339Z   VisualStudioVersion: 17.0
2025-08-05T20:16:51.9070728Z   VS170COMNTOOLS: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\
2025-08-05T20:16:51.9071146Z   VSCMD_ARG_app_plat: Desktop
2025-08-05T20:16:51.9071428Z   VSCMD_ARG_HOST_ARCH: arm64
2025-08-05T20:16:51.9071661Z   VSCMD_ARG_TGT_ARCH: arm64
2025-08-05T20:16:51.9071894Z   VSCMD_VER: 17.14.9
2025-08-05T20:16:51.9072235Z   VSINSTALLDIR: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\
2025-08-05T20:16:51.9072950Z   VSSDK150INSTALL: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VSSDK
2025-08-05T20:16:51.9073443Z   VSSDKINSTALL: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VSSDK
2025-08-05T20:16:51.9074138Z   WindowsLibPath: C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.26100.0;C:\Program Files (x86)\Windows Kits\10\References\10.0.26100.0
2025-08-05T20:16:51.9075013Z   WindowsSdkBinPath: C:\Program Files (x86)\Windows Kits\10\bin\
2025-08-05T20:16:51.9075382Z   WindowsSdkDir: C:\Program Files (x86)\Windows Kits\10\
2025-08-05T20:16:51.9075746Z   WindowsSDKLibVersion: 10.0.26100.0\
2025-08-05T20:16:51.9076142Z   WindowsSdkVerBinPath: C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\
2025-08-05T20:16:51.9076540Z   WindowsSDKVersion: 10.0.26100.0\
2025-08-05T20:16:51.9076983Z   WindowsSDK_ExecutablePath_x64: C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\
2025-08-05T20:16:51.9077639Z   WindowsSDK_ExecutablePath_x86: C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\
2025-08-05T20:16:51.9078092Z   __DOTNET_ADD_64BIT: 1
2025-08-05T20:16:51.9078342Z   __DOTNET_PREFERRED_BITNESS: 64

2025-08-05T20:21:21.7775687Z   C:\Windows\system32\cmd.exe /C "cd /D C:\a\pyside6_qtads\pyside6_qtads\build\temp.win-arm64-cpython-39\Release_PySide6-QtAds && C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-02g1djm3\overlay\Lib\site-packages\shiboken6_generator\shiboken6.exe --generator-set=shiboken --enable-parent-ctor-heuristic --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero --avoid-protected-hack -IC:/a/pyside6_qtads/pyside6_qtads/Qt/6.9.1/msvc2022_arm64/include -IC:/a/pyside6_qtads/pyside6_qtads/Qt/6.9.1/msvc2022_arm64/include/QtCore -IC:/a/pyside6_qtads/pyside6_qtads/Qt/6.9.1/msvc2022_arm64/include -IC:/a/pyside6_qtads/pyside6_qtads/Qt/6.9.1/msvc2022_arm64/include/QtGui -IC:/a/pyside6_qtads/pyside6_qtads/Qt/6.9.1/msvc2022_arm64/include -IC:/a/pyside6_qtads/pyside6_qtads/Qt/6.9.1/msvc2022_arm64/include/QtWidgets -IC:/a/pyside6_qtads/pyside6_qtads/Qt/6.9.1/msvc2022_arm64/include -IC:/a/pyside6_qtads/pyside6_qtads/Qt-Advanced-Docking-System/src -TC:/a/pyside6_qtads/pyside6_qtads -TC:/Users/runneradmin/AppData/Local/Temp/pip-build-env-02g1djm3/overlay/Lib/site-packages/PySide6/typesystems --output-directory=C:/a/pyside6_qtads/pyside6_qtads/build/temp.win-arm64-cpython-39/Release_PySide6-QtAds C:/a/pyside6_qtads/pyside6_qtads/src/bindings.h C:/a/pyside6_qtads/pyside6_qtads/src/bindings.xml"
2025-08-05T20:21:21.7781780Z   qt.shiboken: (bindings) C:\a\pyside6_qtads\pyside6_qtads\src\bindings.xml:461:49: Warning: Duplicate type entry: 'ads::CDockingStateReader'.
2025-08-05T20:21:21.7782702Z   qt.shiboken: (bindings) C:\a\pyside6_qtads\pyside6_qtads\src\bindings.xml:465:45: Warning: Duplicate type entry: 'ads::IFloatingWidget'.
2025-08-05T20:21:21.7783377Z   qt.shiboken: (bindings) CLANG v0.64, builtins includes directory: C:\Program Files\LLVM\lib\clang\20\include
2025-08-05T20:21:21.7784310Z   warning: /Zc:__cplusplus: 'linker' input unused [-Wunused-command-line-argument]
2025-08-05T20:21:21.7784929Z   C:\Program Files\LLVM\lib\clang\20\include\arm_vector_types.h:20:9: error: unknown type name '__mfp8'
2025-08-05T20:21:21.7785521Z   C:\Program Files\LLVM\lib\clang\20\include\arm_vector_types.h:93:24: error: Neon vector size must be 64 or 128 bits
2025-08-05T20:21:21.7786129Z   C:\Program Files\LLVM\lib\clang\20\include\arm_vector_types.h:94:24: error: Neon vector size must be 64 or 128 bits
2025-08-05T20:21:21.7786778Z   C:\Program Files\LLVM\lib\clang\20\include\arm_neon.h:41126:26: error: use of undeclared identifier '__builtin_neon_vcvt1_bf16_mf8_fpm'
...
```

Last working in:
```
2025-07-28T23:16:57.1569532Z Current runner version: '2.326.0'
2025-07-28T23:16:57.1602692Z ##[group]Runner Image Provisioner
2025-07-28T23:16:57.1604153Z Hosted Compute Agent
2025-07-28T23:16:57.1604901Z Version: 20250711.363
2025-07-28T23:16:57.1605539Z Commit: 6785254374ce925a23743850c1cb91912ce5c14c
2025-07-28T23:16:57.1606280Z Build Date: 2025-07-11T20:04:25Z
2025-07-28T23:16:57.1606935Z ##[endgroup]
2025-07-28T23:16:57.1607466Z ##[group]VM Image
2025-07-28T23:16:57.1608078Z - OS: Windows (Arm64)
2025-07-28T23:16:57.1608656Z - Source: Partner
2025-07-28T23:16:57.1609285Z - Name: Microsoft Windows 11 Desktop by Arm Limited
2025-07-28T23:16:57.1609938Z - Version: 20250716.19.1
2025-07-28T23:16:57.1610925Z - Included Software: https://github.com/actions/partner-runner-images/blob/main/images/arm-windows-11-image.md

2025-07-28T23:23:44.1667852Z   Platform: arm64
2025-07-28T23:23:44.1668040Z   UCRTVersion: 10.0.26100.0
2025-07-28T23:23:44.1668314Z   UniversalCRTSdkDir: C:\Program Files (x86)\Windows Kits\10\
2025-07-28T23:23:44.1668768Z   VCIDEInstallDir: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\VC\
2025-07-28T23:23:44.1669249Z   VCINSTALLDIR: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\
2025-07-28T23:23:44.1669683Z   VCPKG_ROOT: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\vcpkg
2025-07-28T23:23:44.1670200Z   VCToolsInstallDir: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.43.34808\
2025-07-28T23:23:44.1670803Z   VCToolsRedistDir: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Redist\MSVC\14.42.34433\
2025-07-28T23:23:44.1671222Z   VCToolsVersion: 14.43.34808
2025-07-28T23:23:44.1671428Z   VisualStudioVersion: 17.0
2025-07-28T23:23:44.1671780Z   VS170COMNTOOLS: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\
2025-07-28T23:23:44.1672155Z   VSCMD_ARG_app_plat: Desktop
2025-07-28T23:23:44.1672365Z   VSCMD_ARG_HOST_ARCH: arm64
2025-07-28T23:23:44.1672569Z   VSCMD_ARG_TGT_ARCH: arm64
2025-07-28T23:23:44.1672763Z   VSCMD_VER: 17.13.1
2025-07-28T23:23:44.1673038Z   VSINSTALLDIR: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\
2025-07-28T23:23:44.1673477Z   VSSDK150INSTALL: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VSSDK
2025-07-28T23:23:44.1673927Z   VSSDKINSTALL: C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VSSDK
2025-07-28T23:23:44.1674548Z   WindowsLibPath: C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.26100.0;C:\Program Files (x86)\Windows Kits\10\References\10.0.26100.0
2025-07-28T23:23:44.1675133Z   WindowsSdkBinPath: C:\Program Files (x86)\Windows Kits\10\bin\
2025-07-28T23:23:44.1675480Z   WindowsSdkDir: C:\Program Files (x86)\Windows Kits\10\
2025-07-28T23:23:44.1675774Z   WindowsSDKLibVersion: 10.0.26100.0\
2025-07-28T23:23:44.1676114Z   WindowsSdkVerBinPath: C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\
2025-07-28T23:23:44.1676462Z   WindowsSDKVersion: 10.0.26100.0\
2025-07-28T23:23:44.1677105Z   WindowsSDK_ExecutablePath_x64: C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\
2025-07-28T23:23:44.1677757Z   WindowsSDK_ExecutablePath_x86: C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\
2025-07-28T23:23:44.1678165Z   __DOTNET_ADD_64BIT: 1
2025-07-28T23:23:44.1678356Z   __DOTNET_PREFERRED_BITNESS: 64

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -39,10 +39,10 @@ jobs:
             pyside_ver: ${{ needs.build_src.outputs.rel0 }}
             wheel: cp39-win_amd64
             arch: x86_64
-          # - os: windows-11-arm
-          #   pyside_ver: ${{ needs.build_src.outputs.rel0 }}
-          #   wheel: cp39-win_arm64
-          #   arch: arm64
+          - os: windows-11-arm
+            pyside_ver: ${{ needs.build_src.outputs.rel0 }}
+            wheel: cp39-win_arm64
+            arch: arm64
           - os: ubuntu-latest
             pyside_ver: ${{ needs.build_src.outputs.rel0 }}
             wheel: cp39-manylinux_x86_64
@@ -70,11 +70,18 @@ jobs:
           if [[ "$RUNNER_OS" == "Windows" ]]; then
             QT_BASE_DIR=${GITHUB_WORKSPACE}\\Qt
             if [[ "${{ matrix.arch }}" == "arm64" ]]; then
-              echo "QT_HOST_PATH=$QT_BASE_DIR\\$QT_VERSION\\msvc2022_64" >> $GITHUB_ENV
               QT6_DIR=$QT_BASE_DIR\\$QT_VERSION\\msvc2022_arm64
+              QT_PLATFORM=windows_arm64
+              QT_ARCH=win64_msvc2022_arm64
             else
               QT6_DIR=$QT_BASE_DIR\\$QT_VERSION\\msvc2022_64
+              QT_PLATFORM=windows
+              QT_ARCH=win64_msvc2022_64
             fi
+            echo "QT_PLATFORM=$QT_PLATFORM" >> $GITHUB_ENV
+            echo "QT_ARCH=$QT_ARCH" >> $GITHUB_ENV
+            echo "CC=cl.exe" >> $GITHUB_ENV
+            echo "CXX=cl.exe" >> $GITHUB_ENV
           elif [[ "$RUNNER_OS" == "macOS" ]]; then
             QT_BASE_DIR=${GITHUB_WORKSPACE}/Qt
             QT6_DIR=$QT_BASE_DIR/$QT_VERSION/macos
@@ -93,6 +100,14 @@ jobs:
         if: runner.os == 'Windows'
         with:
           arch: ${{ matrix.arch }}
+      # XXX: Install headers compatible with shiboken's older libclang
+      #   C:\llvm\lib\clang\20\include\arm_vector_types.h:20:9: error: unknown type name '__mfp8'
+      - name: Install LLVM and Clang
+        if: matrix.os == 'windows-11-arm'
+        uses: KyleMayes/install-llvm-action@98e68e10c96dffcb7bfed8b2144541a66b49aa02 # v2
+        with:
+          version: "19"
+          directory: "C:\\llvm"
       - name: Build wheels
         uses: pypa/cibuildwheel@63fd63b352a9a8bdcc24791c9dbee952ee9a8abc # v3.3.0
         with:
@@ -110,6 +125,7 @@ jobs:
             PIP_EXTRA_INDEX_URL=https://download.qt.io/official_releases/QtForPython/
             VERBOSE=1
           CIBW_ENVIRONMENT_WINDOWS: >
+            CLANG_INSTALL_DIR="${LLVM_PATH}"
             CMAKE_PREFIX_PATH="${QT6_DIR}\\lib\\cmake"
             PIP_EXTRA_INDEX_URL=https://download.qt.io/official_releases/QtForPython/
             VERBOSE=1
@@ -124,7 +140,7 @@ jobs:
             && python3 ./scripts/set-qt-constraint.py ${{ matrix.pyside_ver }}
           CIBW_BEFORE_BUILD_WINDOWS: >
             python3 -m pip install "aqtinstall!=3.2.1"
-            && python3 -m aqt install-qt windows desktop ${{ env.QT_VERSION }} win64_msvc2022_64 --outputdir ${{ env.QT_BASE_DIR }} --base http://mirrors.ocf.berkeley.edu/qt/
+            && python3 -m aqt install-qt ${{ env.QT_PLATFORM }} desktop ${{ env.QT_VERSION }} ${{ env.QT_ARCH }} --outputdir ${{ env.QT_BASE_DIR }} --base http://mirrors.ocf.berkeley.edu/qt/
             && python3 ./scripts/set-qt-constraint.py ${{ matrix.pyside_ver }}
           CIBW_MANYLINUX_X86_64_IMAGE: manylinux_2_34
           CIBW_ARCHS_MACOS: x86_64 arm64

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -248,8 +248,10 @@ target_link_libraries(${bindings_library} PRIVATE ${project_library})
 target_link_libraries(${bindings_library} PRIVATE ${pyside_shared_libraries})
 target_link_libraries(${bindings_library} PRIVATE ${shiboken_module_shared_libraries})
 
+if(NOT WIN32)
 target_compile_options(qtadvanceddocking-qt6 PRIVATE -Og -fPIC)
 target_compile_options(${bindings_library} PRIVATE -Og -fPIC)
+endif()
 target_compile_definitions(${bindings_library} PRIVATE "-DPy_LIMITED_API=0x03090000")
 
 # Adjust the name of generated module.

